docs/develop/typescript/cancellation.mdx

```diff
@@ -383,3 +383,20 @@ export async function shieldAwaitedInRootScope(): Promise<any> {
 ```
 
 <!--SNIPEND-->
+
+## Cancel an Activity from a Workflow {#cancel-an-activity}
+
+Canceling an Activity from within a Workflow requires that the Activity Execution sends Heartbeats and sets a Heartbeat Timeout.
+If the Heartbeat is not invoked, the Activity cannot receive a cancellation request.
+When any non-immediate Activity is executed, the Activity Execution should send Heartbeats and set a [Heartbeat Timeout](/encyclopedia/detecting-activity-failures#heartbeat-timeout) to ensure that the server knows it is still working.
```

Comment:
What is a "non-immediate" Activity? One that is asynchronous? One that doesn't return control and blocks? Long lived? If I were reading this, I'd be confused.